Shared/UserAgent.swift

```diff
@@ -35,33 +35,113 @@ open class UserAgent {
     }
+    
     public static func desktopUserAgent() -> String {
-        return "Mozilla/5.0 (Macintosh; Intel Mac OS X 10.15) AppleWebKit/605.1.15 (KHTML, like Gecko) \(uaBitFx) \(uaBitSafari)"
+        return UserAgentBuilder.defaultDesktopUserAgent().userAgent()
     }
 
     public static func mobileUserAgent() -> String {
-        return "Mozilla/5.0 (\(UIDevice.current.model); CPU OS \(UIDevice.current.systemVersion.replacingOccurrences(of: ".", with: "_")) like Mac OS X) AppleWebKit/605.1.15 (KHTML, like Gecko) \(uaBitFx)  \(uaBitMobile) \(uaBitSafari)"
+        return UserAgentBuilder.defaultMobileUserAgent().userAgent()
     }
-
-    public static func oppositeUserAgent() -> String {
-        let isDefaultUADesktop = UserAgent.isDesktop(ua: UserAgent.defaultUserAgent())
+  
+    public static func oppositeUserAgent(domain: String) -> String {
+        let isDefaultUADesktop = UserAgent.isDesktop(ua: UserAgent.getUserAgent(domain: domain))
         if isDefaultUADesktop {
-            return mobileUserAgent()
+            return UserAgent.getUserAgent(domain: domain, platform: .Mobile)
+        } else {
+            return UserAgent.getUserAgent(domain: domain, platform: .Desktop)
+        }
+    }
+    
+    public static func getUserAgent(domain: String, platform: UserAgentPlatform) -> String {
+        switch platform {
+        case .Desktop:
+            if let customUA = CustomUserAgentConstant.desktopUserAgent[domain] {
+                return customUA
+            } else {
+                return desktopUserAgent()
+            }
+        case .Mobile:
+            if let customUA = CustomUserAgentConstant.mobileUserAgent[domain] {
+                return customUA
+            } else {
+                return mobileUserAgent()
+            }
+        }
+    }
+    
+    public static func getUserAgent(domain: String = "") -> String {
+        // As of iOS 13 using a hidden webview method does not return the correct UA on
+        // iPad (it returns mobile UA). We should consider that method no longer reliable.
+        if UIDevice.current.userInterfaceIdiom == .pad {
+            return getUserAgent(domain: domain, platform: .Desktop)
         } else {
-            return desktopUserAgent()
+            return getUserAgent(domain: domain, platform: .Mobile)
         }
     }
 }
+
+public enum UserAgentPlatform {
+    case Desktop
+    case Mobile
+}
+
+public struct CustomUserAgentConstant {
+    public static let mobileUserAgent = [
+        "whatsapp.com":
+        UserAgentBuilder.defaultMobileUserAgent().userAgent(),
+        "paypal.com":
+        UserAgentBuilder.defaultMobileUserAgent().userAgent(),
+        "yahoo.com":
+        UserAgentBuilder.defaultMobileUserAgent().userAgent()]
+    public static let desktopUserAgent = [
+        "whatsapp.com":
+        UserAgentBuilder.defaultDesktopUserAgent().clone(extensions: "Version/\(AppInfo.appVersion) \(UserAgent.uaBitSafari)"),
```

Comment:
Nit: move this to a constant above as it is used 3x in this class, and more will be added

Reply:
Moved to constant above :)